Formula/asdf.rb

```diff
@@ -26,12 +26,7 @@ def install
     libexec.install Dir["*"]
     touch libexec/"asdf_updates_disabled"
 
-    # TODO: Remove these placeholders on 31 August 2022
     bin.write_exec_script libexec/"bin/asdf"
```

Comment:
I think this one should go too?

Reply:
@carlocab, I did try removing it, but without that line, we don't end up with an `asdf` in our PATH at all. From my original post:

> You may note that the TODO made it look like we should remove the bin.write_exec_script libexec/"bin/asdf" as well. I left that in because without it, we do not end up with an asdf binary in our path at all. I don't know enough about Homebrew to know if this is actually the right way to get a binary on the PATH or not.

Reply:
Yes, that is intentional: https://github.com/Homebrew/homebrew-core/pull/81664#discussion_r680186510

Reply:
Ah! That's a really interesting discussion, and I totally understand why you'd like to remove `asdf` from the PATH to force people to set up asdf "correctly".

> if there truly cannot be a user with a use-case for doing brew install asdf then calling asdf for some reason without sourcing the setup scripts

I actually am a maintainer of [asdf-direnv](https://github.com/asdf-community/asdf-direnv), which provides a shimless alternative way of using asdf, and our tool *does* expect a regular old `asdf` binary on the `PATH` (*not* a shell function). So I'd ask that to avoid breaking things for our users that homebrew continue to provide `asdf` as a regular binary on the PATH.

Reply:
@ericbn, thoughts?

Reply:
I know the asdf installation instructions with homebrew, and the ohmyzsh plugin have been updated to use the libexec subdirectory. So it's looking good from their perspectives.

And yes, that line should go away too. The new way is to source /usr/local/opt/asdf/libexec/asdf.sh directly, not to source /usr/local/opt/asdf/asdf.sh (which sources the former file).

Reply:
This breaks asdf-direnv though, as stated above.

Reply:
Sorry, I missed that comment.

@jfly, the `ASDF_DIR` environment variable in the homebrew installation is set to `/usr/local/opt/asdf/libexec`, and `$ASDF_DIR/bin` is added to the path. So the `/usr/local/opt/asdf/libexec/bin/asdf` executable will be used. This setup is done when sourcing the `/usr/local/opt/asdf/libexec/asdf.sh` file. This should be good for asdf-direnv, right?

Reply:
> This setup is done when sourcing the /usr/local/opt/asdf/libexec/asdf.sh file. This should be good for asdf-direnv, right?

You are right that this would work for asdf-direnv users who source that `asdf.sh` file. However, many users (myself included) do *not* source an `asdf.sh` script: they just have an asdf script (not a shell function!) in their `PATH` and direnv configured, and that's it.

> EDIT: The asdf-direnv pro-tips should also work as expected with the above file removed, since it's referring to ASDF_DIR, which should be /usr/local/opt/asdf/libexec in case of a homebrew installation.

That's correct, again assuming people source that `asdf.sh` script, but not all asdf-direnv users *do* source that script.

It looks like @carlocab merged this PR up as is with an asdf script still on the `PATH`. so speaking purely as an asdf-direnv maintainer, I'm happy with the state things are in.